error: expected `=`
 --> $DIR/invalid.rs:6:24
  |
6 |         const ASSOC: u8; //~ ERROR expected `=`
  |                        ^

error: expected `=`
  --> $DIR/invalid.rs:10:19
   |
10 |         type Assoc; //~ ERROR expected `=`
   |                   ^

error: expected `{`
  --> $DIR/invalid.rs:14:19
   |
14 |         fn assoc(); //~ ERROR expected `{`
   |                   ^

error: expected one of: `default`, `fn`, `const`, `type`
  --> $DIR/invalid.rs:19:9
   |
19 |         mac!(); //~ ERROR expected one of: `default`, `fn`, `const`, `type`
   |         ^^^

error: unexpected token: `,`
  --> $DIR/invalid.rs:23:19
   |
23 |     #[ext(ExtraArg,)] //~ ERROR unexpected token: `,`
   |                   ^

error: use `pub impl` instead
  --> $DIR/invalid.rs:26:11
   |
26 |     #[ext(pub OldVisSyntax1)] //~ ERROR use `pub impl` instead
   |           ^^^

error: use `pub(crate) impl` instead
  --> $DIR/invalid.rs:29:11
   |
29 |     #[ext(pub(crate) OldVisSyntax2)] //~ ERROR use `pub(crate) impl` instead
   |           ^^^^^^^^^^

error: all associated items must have a visibility of `pub`
  --> $DIR/invalid.rs:39:15
   |
39 |         const ASSOC2: u8 = 2; //~ ERROR all associated items must have a visibility of `pub`
   |               ^^^^^^

error: all associated items must have inherited visibility
  --> $DIR/invalid.rs:46:9
   |
46 |         pub fn assoc2(&self) {} //~ ERROR all associated items must have inherited visibility
   |         ^^^

error: all associated items must have a visibility of `pub(crate)`
  --> $DIR/invalid.rs:52:9
   |
52 |         pub type Assoc2 = (); //~ ERROR all associated items must have a visibility of `pub(crate)`
   |         ^^^

error: all associated items must have inherited visibility
  --> $DIR/invalid.rs:59:9
   |
59 |         pub fn assoc2(&self) {} //~ ERROR all associated items must have inherited visibility
   |         ^^^
