error: expected `=`
 --> tests/ui/invalid.rs:8:24
  |
8 |         const ASSOC: u8; //~ ERROR expected `=`
  |                        ^

error: expected `=`
  --> tests/ui/invalid.rs:12:19
   |
12 |         type Assoc; //~ ERROR expected `=`
   |                   ^

error: expected `{`
  --> tests/ui/invalid.rs:16:19
   |
16 |         fn assoc(); //~ ERROR expected `{`
   |                   ^

error: expected one of: `default`, `fn`, `const`, `type`
  --> tests/ui/invalid.rs:21:9
   |
21 |         mac!(); //~ ERROR expected one of: `default`, `fn`, `const`, `type`
   |         ^^^

error: unexpected token: `,`
  --> tests/ui/invalid.rs:25:19
   |
25 |     #[ext(ExtraArg,)] //~ ERROR unexpected token
   |                   ^

error: use `pub impl` instead
  --> tests/ui/invalid.rs:28:11
   |
28 |     #[ext(pub OldVisSyntax1)] //~ ERROR use `pub impl` instead
   |           ^^^

error: use `pub(crate) impl` instead
  --> tests/ui/invalid.rs:31:11
   |
31 |     #[ext(pub(crate) OldVisSyntax2)] //~ ERROR use `pub(crate) impl` instead
   |           ^^^^^^^^^^

error: all associated items must have a visibility of `pub`
  --> tests/ui/invalid.rs:41:15
   |
41 |         const ASSOC2: u8 = 2; //~ ERROR all associated items must have a visibility of `pub`
   |               ^^^^^^

error: all associated items must have inherited visibility
  --> tests/ui/invalid.rs:48:9
   |
48 |         pub fn assoc2(&self) {} //~ ERROR all associated items must have inherited visibility
   |         ^^^

error: all associated items must have a visibility of `pub(crate)`
  --> tests/ui/invalid.rs:54:9
   |
54 |         pub type Assoc2 = (); //~ ERROR all associated items must have a visibility of `pub(crate)`
   |         ^^^

error: all associated items must have inherited visibility
  --> tests/ui/invalid.rs:61:9
   |
61 |         pub fn assoc2(&self) {} //~ ERROR all associated items must have inherited visibility
   |         ^^^
