error[E0603]: trait `StrExt1` is private
  --> $DIR/visibility-2.rs:34:14
   |
34 |     use foo::StrExt1; //~ ERROR trait `StrExt1` is private [E0603]
   |              ^^^^^^^ private trait
   |
note: the trait `StrExt1` is defined here
  --> $DIR/visibility-2.rs:5:5
   |
5  |     impl str {
   |     ^^^^^^^^

error[E0603]: trait `StrExt2` is private
  --> $DIR/visibility-2.rs:36:14
   |
36 |     use foo::StrExt2; //~ ERROR trait `StrExt2` is private [E0603]
   |              ^^^^^^^ private trait
   |
note: the trait `StrExt2` is defined here
  --> $DIR/visibility-2.rs:11:11
   |
11 |       #[ext(pub(self) StrExt2)]
   |  ___________^
12 | |     impl str {
13 | |         fn method2(&self, pat: &str) -> String {
14 | |             self.replace(pat, "_")
15 | |         }
16 | |     }
   | |_____^

error[E0603]: trait `StrExt3` is private
  --> $DIR/visibility-2.rs:38:19
   |
38 |     use foo::bar::StrExt3; //~ ERROR trait `StrExt2` is private [E0603]
   |                   ^^^^^^^ private trait
   |
note: the trait `StrExt3` is defined here
  --> $DIR/visibility-2.rs:21:15
   |
21 |           #[ext(pub(super) StrExt3)]
   |  _______________^
22 | |         impl str {
23 | |             fn method3(&self, pat: &str) -> String {
24 | |                 self.replace(pat, "_")
25 | |             }
26 | |         }
   | |_________^
